### PR TITLE
[TT-10792] Add disabled flag for version-level request header ops

### PR DESCRIPTION
**Pull request by furkansenharputlu**
Task: https://tyktech.atlassian.net/browse/TT-4851
Subtask: https://tyktech.atlassian.net/browse/TT-10792

### CODE REVIEW

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5874/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7216456608)

**Comment by buger** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5874/merge
Commit: ba5405930b1a62f707f15c492528a11295e776ad 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7216530816)

**Comment by buger** [comment]
**API tests result - postgres15-sha256 env: success** :white_check_mark: 
Branch used: refs/pull/5874/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7216456652)

**Comment by buger** [comment]
**API tests result - postgres15-murmur64 env: success** :white_check_mark: 
Branch used: refs/pull/5874/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7216456652)

**Comment by buger** [comment]
**API tests result - mongo44-murmur64 env: success** :white_check_mark: 
Branch used: refs/pull/5874/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7216456652)

**Comment by buger** [comment]
**API tests result - mongo44-sha256 env: success** :white_check_mark: 
Branch used: refs/pull/5874/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7216456652)

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5874/merge
Commit: e5edb50be83f23abd74cc3983e7a36bab044d2e6 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7216688664)

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5874/merge
Commit: 6ab9461e0e002bf8f834bfa014af826319a690d7 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7216811747)

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5874/merge
Commit: 9a1a7c900738030a8451c3ab7532bf15dcff8312 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7216901851)

File: gateway/mw_modify_headers.go [hunk 1]
```diff
@@ -18,8 +18,7 @@ func (t *TransformHeaders) Name() string {
 func (t *TransformHeaders) EnabledForSpec() bool {
 	for _, version := range t.Spec.VersionData.Versions {
 		if len(version.ExtendedPaths.TransformHeader) > 0 ||
-			len(version.GlobalHeaders) > 0 ||
-			len(version.GlobalHeadersRemove) > 0 {
+			(!version.GlobalHeadersDisabled && (len(version.GlobalHeaders) > 0 || len(version.GlobalHeadersRemove) > 0)) {
```
Comment (titpetric) [inline gateway/mw_modify_headers.go:21]: Can we maybe start solving the negative flags by adding a `*VersionInfo.GlobalHeadersEnabled() bool` function?

**Review by titpetric — APPROVED** [review]

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5874/merge
Commit: 74f1b818951cb4849311db476d211500786a7874 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7222343411)

**Comment by buger** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5874/merge
Commit: bc8db2c747b848d1de1c5631c2e75d0161b3cd12 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7222226290)

**Pull request merged**